enhancements/installer/openstack-supported-versions.md

```diff
@@ -0,0 +1,226 @@
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [Design Details](#design-details)
+    - [User Stories](#user-stories)
+      - [Check the OpenStack version](#check-the-openstack-version)
+      - [Ignore the version check](#ignore-the-version-check)
+        - [Example Usage](#example-usage)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+      - [Dev Preview -> Tech Preview](#dev-preview---tech-preview)
+      - [Tech Preview -> GA](#tech-preview---ga)
+    - [Infrastructure Needs](#infrastructure-needs)
+    - [Drawbacks](#drawbacks)
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+OpenShift is not supported when running on top of an unsupported verison
+of OpenStack.
+While the [Support Matrix](https://access.redhat.com/articles/4679401) is
+well documented, OpenShift doesn't check which version of
+OpenStack is running before attempting an operation (e.g. installation or
+upgrade). 
+Running OpenShift over an unsupported version of OpenStack is problematic
+and can lead to unstable infrastructure. To avoid that, it is proposed
+to programmatically verify the version of OpenStack is supported
+otherwise stop an installation or block and upgrade.
+
+## Motivation
+
+### Goals
+
+- Disallow the installation or upgrade of OpenShift done via the some
+  operator condition or check to run against a cloud where the OpenStack
+  version is not supported.
+- Allow an administrator to ignore the OpenStack version check and force the
+  installer to proceed anyway. This can be useful if the OpenStack cloud
+  is partially upgraded to a major version or if there is a Support
+  Exception granted to run that deprecated version of OpenStack.
+
+### Non-Goals
+
+- Backport this feature to stable releases.
+
+## Proposal
+
+In order to implement this feature fully, the following changes must be made:
+
+- Update gophercloud vendoring to (at least) [b56bf1](https://github.com/gophercloud/gophercloud/pull/2037/commits/5a1daf082451587459058cc539c53f2ae1aaaef3)
+  so we can benefit from the apiversions package.
+- Create a constant that will list all the supported versions
+  of Compute API based on the upstream manual.
+- The implementer might decide to use a regexp instead of an array.
+- In the openshift-installer, add a new validation against the OpenStack
+  cloud to make sure that the Compute API version is supported.
+- Create a configuration flag `ignoreVersionCheck` that when set to
+  true will send a warning if the OpenStack version isn't supported and
+  proceed anyway.
+- The upgrade of OpenShift should warn an admin about the incompatibility
```

Comment:
Like with installs we should provide the user / admin the ability to ignore and proceed.
However it a user/admin does this we should record that they did this (IE: record that they forced the upgrade).